### PR TITLE
Fix Issue 23424 - improve error when template instantiation doesn't match an overload

### PR DESCRIPTION
**Pull request by ntrel**
Show potential candidates.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: compiler/test/fail_compilation/templateoverload.d [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/templateoverload.d(17): Error: template `templateoverload.T` does not match any template declaration
```
Comment (nordlow) [inline compiler/test/fail_compilation/templateoverload.d:4]: Shouldn't this be

```
`template instance templateoverload.T!(1)` does not match any template declaration
```

?

File: compiler/test/fail_compilation/templateoverload.d [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+fail_compilation/templateoverload.d(17):        Candidates are:
+fail_compilation/templateoverload.d(14):        T(X)
+fail_compilation/templateoverload.d(15):        T()
+fail_compilation/templateoverload.d(22): Error: template `templateoverload.V` does not match any template declaration
```
Comment (nordlow) [inline compiler/test/fail_compilation/templateoverload.d:8]: Shouldn't this be template

```
template instance `templateoverload.T!(int)` does not match any template declaration
```

?

**Comment by nordlow** [comment]
Thanks for doing this, btw.

File: compiler/test/fail_compilation/templateoverload.d (hunk 2 above)
Comment (nordlow) [reply compiler/test/fail_compilation/templateoverload.d:8]: See https://issues.dlang.org/show_bug.cgi?id=23424

File: compiler/test/fail_compilation/templateoverload.d (hunk 1 above)
Comment (ntrel) [reply compiler/test/fail_compilation/templateoverload.d:4]: Thanks, I made https://github.com/dlang/dmd/pull/14604

Comment (nordlow) [reply compiler/test/fail_compilation/templateoverload.d:4]: Awesome!

**Pull request merged**